### PR TITLE
engine: fix up dc deploys when tilt is configured to talk to k8s

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc,

Please review the following commits I made in branch nicks/dcimage4:

10fa555e9e7a75144305b6cc7abee82e56862927 (2019-03-12 21:08:59 -0400)
engine: fix up dc deploys when tilt is configured to talk to k8s

### CODE REVIEW

File: internal/engine/build_and_deployer.go [hunk 1]
```diff
@@ -105,5 +105,5 @@ func DefaultBuildOrder(sbad *SyncletBuildAndDeployer, cbad *LocalContainerBuildA
 	if runtime == container.RuntimeDocker {
 		ibad.SetInjectSynclet(true)
 	}
-	return BuildOrder{sbad, dcbad, ibad}
+	return BuildOrder{sbad, cbad, dcbad, ibad}
```
Comment (maiamcc) [inline internal/engine/build_and_deployer.go]: 😳 good catch!

File: internal/engine/extractors.go [hunk 2]
```diff
@@ -45,6 +45,18 @@ func extractImageTargets(specs []model.TargetSpec) []model.ImageTarget {
 	return iTargets
 }
 
+func extractDockerComposeTargets(specs []model.TargetSpec) []model.DockerComposeTarget {
+	iTargets := make([]model.DockerComposeTarget, 0)
```
Comment (maiamcc) [inline internal/engine/extractors.go]: `iTarget` is shorthand for "image targets", call this something else

File: internal/engine/build_and_deployer_test.go [hunk 3]
```diff
@@ -553,6 +553,27 @@ func TestContainerBuildMultiStage(t *testing.T) {
 	assert.Equal(t, k8s.MagicTestContainerID, result.OneAndOnlyContainerID().String())
 }
 
+func TestDockerComposeFastBuild(t *testing.T) {
+	f := newBDFixture(t, k8s.EnvNone)
```
Comment (maiamcc) [inline internal/engine/build_and_deployer_test.go]: imo this test is extra compelling if you set a k8s Env that would normally do a synclet build, and then asserted that there was no synclet call

File: internal/engine/extractors.go (hunk 2 above)
Comment (nicks) [reply internal/engine/extractors.go]: done

File: internal/engine/build_and_deployer_test.go (hunk 3 above)
Comment (nicks) [reply internal/engine/build_and_deployer_test.go]: done!

**Pull request merged**